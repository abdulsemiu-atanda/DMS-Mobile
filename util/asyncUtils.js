import sendRequest from '../requests/requestUtil'

/**
 * Constructs async actions constants
 * @param {string} baseName
 * @returns {Object}
 */
export const asyncActionNames = baseName => ({
  failure: `${baseName}_FAILURE`,
  loading: `${baseName}_LOADING`,
  success: `${baseName}_SUCCESS`
})

/**
 * Constructs Redux async actions
 * @param {string} actionName
 * @returns {Object}
 */
export const asyncActions = (actionName) => ({
  loading: bool => ({
    type: asyncActionNames(actionName).loading,
    data: bool
  }),
  failure: (bool, error) => ({
    type: asyncActionNames(actionName).failure,
    data: {error, status: bool}
  }),
  success: data => ({
    type: asyncActionNames(actionName).success,
    data
  })
})

/**
 * Dispatches Redux Actions
 * @async asyncRequest
 * @param {string} ACTION_NAME - The name of action to be dispatched
 * @param {string} endpoint - The endpoint to resource
 * @param {string} method - HTTP method
 * @param {Object} data - HTTP request body
 * @param {string} token - Authentication token for request
 * @returns {Function} - Function that dispatches Redux Actions
 */

export const asyncRequest = (ACTION_NAME, endpoint, method, data, token) => dispatch => {
  dispatch(asyncActions(ACTION_NAME).loading(true))
  sendRequest(endpoint, method, data, token)
    .then(res => res.json())
    .then(res => {
      dispatch(asyncActions(ACTION_NAME).success(res))
      dispatch(asyncActions(ACTION_NAME).loading(false))
    })
    .catch(err => dispatch(asyncActions(ACTION_NAME).failure(true, err)))
}
