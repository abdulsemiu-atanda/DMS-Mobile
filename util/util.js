import jwt from 'jwt-decode'
import Immutable from 'immutable'

/**
 * Checks if a token has expired
 * @param {string} token - Valid jwt token
 * @returns {boolean}
 */
export const isTokenExpired = token => {
  const decoded = jwt(token)

  return decoded.exp < Date.now() / 1000
}

/**
 * Makes the first letter of a string capital
 * @param {string} string
 * @returns {string} - String with capital first letter
 */
export const ucFirst = string => `${string.charAt(0).toUpperCase()}${string.slice(1)}`

/**
 * Converts Javascript Object or Array into Immutable Map or List appropriately
 * @param {(Object|Array)} data
 * @returns {(Map|List)} - Immutable map or list
 */
export const makeImmutable = data => Immutable.fromJS(data)

/**
 * Empty funstion for props that takes in functions in test
 */
export const noop = () => {}
