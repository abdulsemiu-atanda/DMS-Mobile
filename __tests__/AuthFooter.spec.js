import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import AuthFooter from '../components/shared/AuthFooter.react'
import {noop} from '../util/util'

const props = {
  flipCard: noop,
  screen: 'Login'
}

describe('AuthFooter', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AuthFooter {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children[0].children[0]).toEqual("DON'T HAVE AN ACCOUNT?")
    expect(tree.children[1].children[0].children[0]).toEqual('SIGN UP')
  })

  it('renders correctly when screen is sign up', () => {
    const newProps = {
      ...props,
      screen: 'SignUp'
    }
    const tree = renderer.create(<AuthFooter {...newProps} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children[0].children[0]).toEqual('ALREADY HAVE AN ACCOUNT?')
    expect(tree.children[1].children[0].children[0]).toEqual('SIGN IN')
  })
})
