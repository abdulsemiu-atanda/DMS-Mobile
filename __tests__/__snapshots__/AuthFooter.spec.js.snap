// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthFooter renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 30,
      },
      Object {
        "height": "16%",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 12,
        "fontWeight": "600",
        "marginRight": 5,
      }
    }
  >
    DON'T HAVE AN ACCOUNT?
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
          "fontWeight": "700",
        }
      }
    >
      SIGN UP
    </Text>
  </View>
</View>
`;

exports[`AuthFooter renders correctly when screen is sign up 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 30,
      },
      Object {
        "height": "16%",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 12,
        "fontWeight": "600",
        "marginRight": 5,
      }
    }
  >
    ALREADY HAVE AN ACCOUNT?
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
          "fontWeight": "700",
        }
      }
    >
      SIGN IN
    </Text>
  </View>
</View>
`;
